contracts/MarketMaker.sol

```diff
@@ -54,43 +59,70 @@ contract MarketMaker is Ownable, IERC1155TokenReceiver {
         collateralToken = _collateralToken;
         conditionId = _conditionId;
         fee = _fee;
+
+        require(collateralToken.transferFrom(tx.origin, this, _funding) && collateralToken.approve(pmSystem, _funding));
+        uint[] memory partition = generateBasicPartition();
+        pmSystem.splitPosition(collateralToken, bytes32(0), conditionId, partition, _funding);
+        funding = _funding;
+
         stage = Stages.MarketCreated;
+        emit AutomatedMarketMakerCreated(funding);
     }
 
     function calcNetCost(int[] outcomeTokenAmounts) public view returns (int netCost);
 
     /// @dev Allows to fund the market with collateral tokens converting them into outcome tokens
     /// @param _funding Funding amount
```

Comment:
Rename `_funding` to match actual behavior of variable. Right now, this suggests that `funding` will be *set to* `_funding`. Try e.g. `fundingChange`.